File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int firstZeroIndex = Integer.MIN_VALUE;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                firstZeroIndex = i;
                break;
            }
        }

        if (firstZeroIndex == Integer.MIN_VALUE || firstZeroIndex == nums.length - 1) {
            return;
        }


        for(int i = firstZeroIndex; i < nums.length ; i++) {
            if(nums[i] != 0) {
               int temp = nums[i];
               nums[firstZeroIndex] = temp;
               nums[i] = 0;
               firstZeroIndex++;
            }
        }        

    }
}

File: 0283-move-zeroes/0283-move-zeroes.kt
class Solution {
    fun moveZeroes(nums: IntArray): Unit {

        if(nums.size < 1) {
            return
        }

        var zeroIndex = Integer.MIN_VALUE

        for(i in 0 until nums.size) {
            if(nums[i] == 0) {
                zeroIndex = i
                break
            }
        }

        if(zeroIndex == Integer.MIN_VALUE) {
            return
        }

        for(i in zeroIndex until nums.size) {
            if(nums[i] != 0) {
                swapValue(nums, zeroIndex, i)        
                zeroIndex++    
            }
        }

    }

    fun swapValue(nums: IntArray, zeroIndex : Int, swappingIndex : Int) {
        // val tempValue = nums[zeroIndex]
        nums[zeroIndex] = nums[swappingIndex]
        nums[swappingIndex] = 0
    }
}